contracts/Joint.sol

```diff
@@ -150,13 +168,28 @@ abstract contract Joint {
 
     function _autoProtect() internal view virtual returns (bool);
 
+    function shouldStartEpoch() public view returns (bool) {
+        // return true if we have balance of A or balance of B while the position is closed
+        return
+            (balanceOfA() > 0 || balanceOfB() > 0) &&
```

Comment:
@fp-crypto can you confirm this? I want to be sure I'm thinking about it in the right way.

First epoch: both providers will be called and they will send their allocation to Joint, then start the epoch. 

Subsequent epochs: if the providers have not been set to 0 debt ratio, then the same harvest that is closing the epoch (only one provider harvested, both providers receive their "loose" tokens) will send the funds back to the joint during adjustPosition. This means that then shouldStartEpoch will return true and the other provider will be harvested. This provider will then send the funds to the joint AND start the new epoch. 

^^ I think we need to stop the provider that finished the epoch (and sent the tokens back to the joint) to return true in harvestTrigger if its balance is 0, otherwise keepers may get confused and harvest that provider several times.

Reply:
Yes, i think this is fine. Agree that the harvest/tend trigger needs to be modded.